src/NzbDrone.Core/Download/Clients/rTorrent/RTorrent.cs

```diff
@@ -0,0 +1,217 @@
+                    }
+
+                    var item = new DownloadClientItem();
+                    item.DownloadClient = Definition.Name;
+                    item.Title = torrent.Name;
+                    item.DownloadId = torrent.Hash;
+                    item.OutputPath = _remotePathMappingService.RemapRemoteToLocal(Settings.Host, new OsPath(torrent.Path));
+                    item.TotalSize = torrent.TotalSize;
+                    item.RemainingSize = torrent.RemainingSize;
+                    item.Category = torrent.Category;
+
+                    if (torrent.DownRate > 0) {
+                        var secondsLeft = torrent.RemainingSize / torrent.DownRate;
+                        item.RemainingTime = TimeSpan.FromSeconds(secondsLeft);
+                    }
+
+                    if (torrent.IsFinished) item.Status = DownloadItemStatus.Completed;
+                    else if (torrent.IsActive) item.Status = DownloadItemStatus.Downloading;
+                    else if (!torrent.IsActive) item.Status = DownloadItemStatus.Paused;
+                    else if (!torrent.IsOpen) item.Status = DownloadItemStatus.Queued;
+
+                    // Since we do not know the user's intent, do not let the Sonarr to remove torrent
+                    item.IsReadOnly = true;
+
+                    items.Add(item);
+                }
+
+                return items;
+            }
+            catch (DownloadClientException ex)
+            {
+                _logger.ErrorException(ex.Message, ex);
+                return Enumerable.Empty<DownloadClientItem>();
+            }
+
+        }
+
+        public override void RemoveItem(string downloadId, bool deleteData)
+        {
+            _proxy.RemoveTorrent(downloadId, Settings);
+
+            if (deleteData)
+            {
+                DeleteItemData(downloadId);
+            }
+        }
+
+        public override DownloadClientStatus GetStatus()
+        {
+            // XXX: This function's correctness has not been considered
+
+            var status = new DownloadClientStatus
+            {
+                IsLocalhost = Settings.Host == "127.0.0.1" || Settings.Host == "localhost"
+            };
+
+            return status;
+        }
+
+        protected override void Test(List<ValidationFailure> failures)
+        {
+            failures.AddIfNotNull(TestConnection());
+            if (failures.Any()) return;
+            failures.AddIfNotNull(TestGetTorrents());
+        }
+
+        private ValidationFailure TestConnection()
+        {
+            try
+            {
+                var version = _proxy.GetVersion(Settings);
+
+                if (new Version(version) < new Version("0.9.0"))
+                {
+                    return new ValidationFailure(string.Empty, "RTorrent version should be at least 0.9.0. Version reported is {0}", version);
```

Comment:
Mixed: http://rakshasa.github.io/rtorrent/

The official one seems to be rTorrent.